### PR TITLE
[UI 1659] - Add File Upload feature to React Native

### PR DESCRIPTION
**Pull request by piotr-suwala**
https://user-images.githubusercontent.com/112620304/208088362-8d6bf74c-0c30-4215-a697-46117b68eb3a.mov

### CODE REVIEW

File: samples/react-native/getting-started/src/getting-started-react-native.tsx [hunk 1]
```diff
@@ -8,8 +8,8 @@ import { Chat, MessageList, MessageInput } from "@pubnub/react-native-chat-compo
 /* Creates and configures your PubNub instance. Be sure to replace "myPublishKey" and "mySubscribeKey"
   with your own keyset. If you wish, modify the default "myFirstUser" uuid value for the chat user. */
 const pubnub = new PubNub({
-  publishKey: "myPublishKey",
-  subscribeKey: "mySubscribeKey",
+  publishKey: "pub-c-0457cb83-0786-43df-bc70-723b16a6e816",
```
Comment (Salet) [inline samples/react-native/getting-started/src/getting-started-react-native.tsx:11]: We need to roll all of the changes in this app back

File: packages/react-native/src/message-list/message-list.style.ts [hunk 2]
```diff
@@ -19,6 +23,10 @@ const darkColors = {
   unreadColor: "#2a2a39",
   reactionBorder: "#28293d",
   reactionActiveBackground: "rgba(239, 58, 67, 0.3)",
+  downloadedSuccessBannerBackground: "rgba(224, 224, 224, 0.9)",
```
Comment (Salet) [inline packages/react-native/src/message-list/message-list.style.ts:26]: I think the colors here should be the other way around. Currently the modal has white background on dark mode and almost black on light mode

File: samples/react-native/getting-started/app.config.js [hunk 3]
```diff
@@ -30,4 +32,14 @@ export default {
     REACT_APP_PUB_KEY: process.env.REACT_APP_PUB_KEY,
     REACT_APP_SUB_KEY: process.env.REACT_APP_SUB_KEY,
   },
+  plugins: [
+    [
+      "expo-image-picker",
```
Comment (Salet) [inline samples/react-native/getting-started/app.config.js:37]: Same here. We probably don't want to clutter the getting started app with a non-essential feature. Perhaps we could move these to the Group Chat app

File: packages/react-native/mock/file-mocks.tsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  launchImageLibraryAsync: jest.fn(() =>
+    Promise.resolve({
+      canceled: false,
+      assets: [{ uri: "some_uri", fileName: "fileName" }],
+    })
+  ),
+};
+
+export const EXPO_DOCUMENT_PICKER_MOCK = {
+  getDocumentAsync: jest.fn(() =>
```
Comment (Salet) [inline packages/react-native/mock/file-mocks.tsx:54]: These are nifty, good job

File: packages/react-native/src/message-input/message-input.tsx [hunk 5]
```diff
@@ -1,30 +1,109 @@
...
+import React, { FC, useState } from "react";
 import { View, Image, TouchableOpacity, TextInput, Animated } from "react-native";
 import { CommonMessageInputProps, useMessageInputCore } from "@pubnub/common-chat-components";
+import { getDocumentAsync } from "expo-document-picker";
 import { useStyle, useRotation } from "../helpers";
 import createDefaultStyle, { MessageInputStyle } from "./message-input.style";
 import AirplaneIcon from "../icons/airplane.png";
 import AirplaneActiveIcon from "../icons/airplaneActive.png";
 import SpinnerIcon from "../icons/spinnerActive.png";
+import FileIcon from "../icons/file.png";
```
Comment (Salet) [inline packages/react-native/src/message-input/message-input.tsx:10]: I think a 2x resolution (40x40) with these icons would've sufficed, but ok ;)

**Review by Salet — COMMENTED** [review]
Thanks for the changes so far, great work overall

File: samples/react-native/getting-started/app.config.js (hunk 3 above)
Comment (piotr-suwala) [reply samples/react-native/getting-started/app.config.js:37]: Actually now when I test that it seems that if I completely remove this section, expo-image-picker ends up with default values here. I'll remove it.

**Review by Salet — APPROVED** [review]
It looks very well now, great work

**Pull request merged**